sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/BoundAttribute.scala

```diff
@@ -23,6 +23,15 @@ import org.apache.spark.sql.catalyst.errors.attachTree
 import org.apache.spark.sql.catalyst.expressions.codegen.{CodeGenContext, GeneratedExpressionCode}
 import org.apache.spark.sql.types._
 
+object BoundReference {
+  def apply(
```

Comment:
Are you sure you want the name parameter to be optional? The bug in the original JIRA might pop up again if there's another code path that produces instances of BoundReference without name metadata in the plan.